### PR TITLE
TDE-2: lundi-mardi of the pre-merge

### PR DESCRIPTION
**Pull request by tde-banana-odoo**

### CODE REVIEW

File: addons/knowledge/models/knowledge_article.py [hunk 1]
```diff
@@ -748,44 +755,52 @@ def _get_max_sequence_inside_parent(self, parent_id):
...
+                    'permission': 'write'
+                })]
+
+        # User cannot write on members, sudo is needed to allow to create a private article
+        # due to access to members
+        if is_private and self.env.user.has_group('base.group_user'):
+            ArticleEnv = self.sudo()
+        else:
+            ArticleEnv = self
+        return ArticleEnv.create(values).with_env(self.env)
```
Comment (dbeguin) [inline addons/knowledge/models/knowledge_article.py]: dafuck ?

File: addons/knowledge/models/knowledge_article.py [hunk 2]
```diff
@@ -553,7 +573,23 @@ def create(self, vals_list):
...
+        sudo_articles = iter(super(Article, self.sudo()).create([
+            vals for vals, can_sudo in zip(vals_list, vals_as_sudo)
+            if can_sudo
+        ]).with_env(self.env))
+        articles = self.env['knowledge.article']
+        for vals, is_sudo in zip(vals_list, vals_as_sudo):
+            if is_sudo:
+                articles += next(sudo_articles)
+            else:
+                articles += next(notsudo_articles)
```
Comment (dbeguin) [inline addons/knowledge/models/knowledge_article.py:561]: :exploding_head:

Comment (dbeguin) [reply addons/knowledge/models/knowledge_article.py:561]: are you sure we keep the order ?

File: addons/knowledge/models/knowledge_article.py [hunk 3]
```diff
@@ -587,8 +587,11 @@ def write(self, vals):
         _resequence = False
         if vals.get('parent_id'):
             parent = self.browse(vals['parent_id'])
-            parent.check_access_rights('write')
-            parent.check_access_rule('write')
+            try:
+                parent.check_access_rule('write')
+            except AccessError:
+                raise AccessError(_("You cannot move an article under %(parent_name)s as you cannot write on it",
+                                    parent_name=parent.display_name))
```
Comment (dbeguin) [inline addons/knowledge/models/knowledge_article.py]: parent.check_access_rights('write') not needed anymore ?

File: addons/knowledge/models/knowledge_article.py (hunk 1 above)
Comment (tde-banana-odoo) [reply addons/knowledge/models/knowledge_article.py]: fixed :D (somehow)

File: addons/knowledge/models/knowledge_article.py (hunk 2 above)
Comment (tde-banana-odoo) [reply addons/knowledge/models/knowledge_article.py:561]: That's the purpose indeed :)

File: addons/knowledge/models/knowledge_article.py (hunk 3 above)
Comment (tde-banana-odoo) [reply addons/knowledge/models/knowledge_article.py]: Yup, issue in rebase, fixed

File: addons/knowledge/controllers/main.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ def article_invite(self, member_id, invitation_hash):
     def redirect_to_article(self, article_id):
         """ This route will redirect internal users to the backend view of the article and the share users to the
         frontend view instead."""
-        article = self._fetch_article(article_id)
+        article = self._fetch_article(article_id, 'read')
```
Comment (dbeguin) [inline addons/knowledge/controllers/main.py:49]: would name the parameter !

**Comment by tde-banana-odoo** [comment]
Should be in master-knowledge, closing.

**Pull request closed without merging**